### DISCUSSION TITLE
How to programatically determine whether signatures are LTV-enabled?

### DISCUSSION THREAD

**Discussion by ofcaah** [MatthiasValvekens/pyHanko#344] [Q&A] [answered]
I don't think there's anything about this in validate_pdf_signature function.

**Comment by MatthiasValvekens** [comment, marked as answer]
(On mobile right now, so lacking in details)

If your target is "will Acrobat consider this signature LTV-enabled?" then that's hard to answer, since those criteria are not public knowledge and could change at any time.

If you're looking for a more vague "will a future validator still be able to verify this signature?" type of check: that's also quite difficult, but more doable.

- The PADES_LTA profile setting in the validation function you're referencing is very opinionated and probably more strict than Acrobat, but it also represents a flawed understanding of the underlying problem. It will be deprecated as soon as the AdES validation API matures.
- That new API will include a function in the next release to simulate future validation to (hopefully) evaluate the future-proofness of the signature more accurately. It's already merged into the master branch in case you'd like to experiment.

**Comment by ofcaah** [comment]
Thank you for the pointers; any indisputable way of proving, that signing certificate was valid and not revoked at verification time is fine; adobe reader agreeing with this is a nice bonus, but not mandatory. Thanks, I'll look into master branch.
